bftengine/include/bftengine/ControlStateManager.hpp

```diff
@@ -14,19 +14,53 @@
 #pragma once
 #include <optional>
 #include "IStateTransfer.hpp"
+#include "ReservedPages.hpp"
+#include "Serializable.h"
+#include "SysConsts.hpp"
 
 namespace bftEngine {
 
-class ControlStateManager {
+static constexpr uint32_t ControlHandlerStateManagerReservedPagesIndex = 1;
+static constexpr uint32_t ControlHandlerStateManagerNumOfReservedPages = 1;
+class ControlStateManager : public ResPagesClient<ControlStateManager,
+                                                  ControlHandlerStateManagerReservedPagesIndex,
+                                                  ControlHandlerStateManagerNumOfReservedPages> {
  public:
-  void setStopAtNextCheckpoint();
-  std::optional<uint64_t> getStopCheckpointToStopAt();
+  void setStopAtNextCheckpoint(int64_t currentSeqNum);
+  std::optional<int64_t> getCheckpointToStopAt();
 
-  ControlStateManager(IStateTransfer& state_transfer);
+  ControlStateManager(IStateTransfer* state_transfer, uint32_t sizeOfReservedPages);
   ControlStateManager& operator=(const ControlStateManager&) = delete;
   ControlStateManager(const ControlStateManager&) = delete;
+  ~ControlStateManager() {}
 
  private:
-  IStateTransfer& state_transfer_;
+  IStateTransfer* state_transfer_;
+  const uint32_t sizeOfReservedPage_;
+  std::string scratchPage_;
+
+  // reserved page indexer
+  struct reservedPageIndexer {
+    uint32_t update_reserved_page_ = 0;
+  };
+
+  reservedPageIndexer reserved_pages_indexer_;
+};
+
+namespace controlStateMessages {
+
+// control state messages
+class StopAtNextCheckpointMessage : public concord::serialize::SerializableFactory<StopAtNextCheckpointMessage> {
+  const std::string getVersion() const override { return "1"; }
```

Comment:
Why a string vs an int?

Reply:
This is the interface of the concord::serialize